### ISSUE TITLE
allow pull right section in collapsable nav bars

### ISSUE THREAD

**Issue by Licenser** [racehub/om-bootstrap#83]
Bootstrap allows right / left aligned nav bars. Currently it seems not possible since the `nav` component handling the bars won't allow multiple `ul` elements inside the collapsable `div`. The following code introduces a `:right` keyword in the nav children that splits the children in half when `colapsable?` is set to true and creates two seperate `ul` elements for those one set to right one set to left:

```clojure
(defcomponentk nav* [owner]
  (render
   [_]
   (let [{:keys [opts children]} (om/get-props owner)
         [bs props] (t/separate Nav opts {:expanded? true
                                          :bs-class "nav"})
         classes {:navbar-collapse (:collapsible? bs)
                  :collapse (not (:expanded? bs))
                  :in (:expanded? bs)}
         ul-props {:ref "ul"
                   :class (d/class-set
                           (merge (t/bs-class-set bs)
                                  {:nav-stacked (:stacked? bs)
                                   :nav-justified (:justified? bs)
                                   :navbar-nav (:navbar? bs)
                                   :pull-right (:pull-right? bs)}))}
         ul-props-left {:ref "ul"
                   :class (d/class-set
                           (merge (t/bs-class-set bs)
                                  {:nav-stacked (:stacked? bs)
                                   :nav-justified (:justified? bs)
                                   :navbar-nav (:navbar? bs)
                                   :pull-right false}))}
         ul-props-right
         {:ref "ul"
          :class (d/class-set
                  (merge (t/bs-class-set bs)
                         {:nav-stacked (:stacked? bs)
                          :nav-justified (:justified? bs)
                          :navbar-nav (:navbar? bs)
                          :pull-right true}))}
         [children-left _ children-right] (partition-by #(= :right %) children)
         children-left (map (clone-nav-item opts) children-left)
         children-right (map (clone-nav-item opts) children-right)]
     (if (and (:navbar? bs)
              (not (:collapsible? bs)))
       (d/ul (u/merge-props props ul-props) children)
       (d/nav (u/merge-props props {:class (d/class-set classes)})
              (if (not (empty? children-left)) (d/ul ul-props-left children-left))
              (if (not (empty? children-right))(d/ul ul-props-right children-right)))))))
```

**Comment by Licenser** [comment]
The result looks like this:
```clojure
(nav
  {:collapsible? true}
    (n/nav-item {:key 1 :href "#/a"} "A")
    (n/nav-item {:key 2 :href "#/b"} "B")
    (n/nav-item {:key 3 :href "#/c"} "C")
    :right
    (n/nav-item {:key 1 :on-click #(conf/logout)} "Logout")))
```

**Comment by Licenser** [comment]
This is a lot cleaner:
```clojure
(defcomponentk nav* [owner]
  (render
   [_]
   (let [{:keys [opts children]} (om/get-props owner)
         [bs props] (t/separate Nav opts {:expanded? true
                                          :bs-class "nav"})
         classes {:navbar-collapse (:collapsible? bs)
                  :collapse (not (:expanded? bs))
                  :in (:expanded? bs)}]
     (if (and (:navbar? bs)
              (not (:collapsible? bs)))
       (let [children (map (clone-nav-item opts) children)
             ul-props {:ref "ul"
                       :class (d/class-set
                               (merge (t/bs-class-set bs)
                                      {:nav-stacked (:stacked? bs)
                                       :nav-justified (:justified? bs)
                                       :navbar-nav (:navbar? bs)
                                       :pull-right (:pull-right? bs)}))}]
         (d/ul (u/merge-props props ul-props) children))
       (let [ul-props-left {:ref "ul"
                            :class (d/class-set
                                    (merge (t/bs-class-set bs)
                                           {:nav-stacked (:stacked? bs)
                                            :nav-justified (:justified? bs)
                                            :navbar-nav (:navbar? bs)
                                            :pull-right false}))}
             ul-props-right
             {:ref "ul"
              :class (d/class-set
                      (merge (t/bs-class-set bs)
                             {:nav-stacked (:stacked? bs)
                              :nav-justified (:justified? bs)
                              :navbar-nav (:navbar? bs)
                              :pull-right true}))}
             [children-left _ children-right] (partition-by #(= :right %) children)
             children-left (map (clone-nav-item opts) children-left)
             children-right (map (clone-nav-item opts) children-right)]
         (d/nav (u/merge-props props {:class (d/class-set classes)})
                (if (not (empty? children-left)) (d/ul ul-props-left children-left))
                (if (not (empty? children-right))(d/ul ul-props-right children-right))))))))
```

**Comment by dpetrovics** [comment]
Hi Heinz, thanks - I can see how this might be useful. Would you might submitting a PR with a brief example for the documentation site?

**Comment by dpetrovics** [comment]
Thanks for this and sorry for the big delay. Just released to Clojars as `0.6.1` along with some other major dependency upgrades in `0.6.0`

**Comment by Licenser** [comment]
No worries, I do OSS myself I know how things go :) thumbs up on the new Clojars and thanks for all the hard work mate!!
